src/map.react.js

```diff
@@ -303,6 +305,20 @@ export default class MapGL extends Component {
     map.addSource(update.id, newSource);
   }
 
+  // Hover and click only query layers whose interactive property is true
+  // If no interactivity is specified, query all layers
+  _updateQueryParams(mapStyle) {
+    if (mapStyle instanceof Immutable.Map) {
+      mapStyle = mapStyle.toJS();
+    }
+
+    const interactiveLayerIds = Array.isArray(mapStyle.layers) ? mapStyle.layers
+      .filter(l => l.interactive)
+      .map(l => l.id) : [];
+    this._queryParams = interactiveLayerIds.length === 0 ? {} :
```

Comment:
We only perform the query if props.onHoverFeatures or props.onClickFeatures is defined. So maybe it's not unreasonable to assume the user wants something to be interactive...?